chromium_src/third_party/blink/renderer/modules/plugins/dom_plugin_array.cc

```diff
@@ -61,36 +38,15 @@ void FarblePlugins(DOMPluginArray* owner,
     case BraveFarblingLevel::OFF: {
       break;
     }
+    case BraveFarblingLevel::BALANCED:
     case BraveFarblingLevel::MAXIMUM: {
-      dom_plugins->clear();
-      // "Maximum" behavior is clear existing plugins + "balanced" behavior,
-      // so fall through here.
-      U_FALLTHROUGH;
-    }
-    case BraveFarblingLevel::BALANCED: {
-      std::mt19937_64 prng = BraveSessionCache::From(*(frame->DomWindow()))
-                                 .MakePseudoRandomGenerator();
-      // The item() method will populate plugin info if any item of
-      // |dom_plugins_| is null, but when it tries, it assumes the
-      // length of |dom_plugins_| == the length of the underlying
-      // GetPluginData()->Plugins(). Once we add our fake plugins, that
-      // assumption will break and the item() method will crash with an
-      // out-of-bounds array access. Rather than patch the item() method, we
-      // ensure that the cache is fully populated now while the assumptions
-      // still hold, so the problematic code is never executed later.
-      for (unsigned index = 0; index < dom_plugins->size(); index++) {
-        auto plugin = frame->GetPluginData()->Plugins()[index];
-        String name = plugin->Name();
-        // Built-in plugins get their names and descriptions farbled as well.
-        if ((name == "Chrome PDF Plugin") || (name == "Chrome PDF Viewer")) {
-          plugin->SetName(PluginReplacementName(&prng));
-          plugin->SetFilename(
-              BraveSessionCache::From(*(frame->DomWindow()))
-                  .GenerateRandomString(plugin->Filename().Ascii(), 32));
-        }
-        (*dom_plugins)[index] =
-            MakeGarbageCollected<DOMPlugin>(frame->DomWindow(), *plugin);
-      }
+      // Both "balanced" and "maximum" behaviors will return a list with 2 fake
```

Comment:
@pes10k is the plugins farbling still relevant with this change?

Reply:
I'd still wait for @pes10k to reply but for the time being just a comment on that my understanding is that farbling is still relevant even if navigators.plugins is now empty because it still adds noise that will help towards reducing the chances of successfully fingerprinting an user session.

Reply:
Yes, seconding @mariospr , we should continue farbling plugins as long as fingerprinters check it, since the randomization provides umbrella protections over other APIs we don't cover